test/engine/SpatialJoinTest.cpp

```diff
@@ -0,0 +1,1957 @@
+      if (lon == -90 || lon == 90) {
+        // only add one point for the poles
+        addPoint(&kg, lon, 0);
+        break;
+      }
+
+      addPoint(&kg, lon, lat);
+
+      if (!verbose) {
+        if (lon % 2 == 1 || (lat > -160 && lat < -20) ||
+            (lat > 20 && lat < 160)) {
+          continue;
+        }
+      }
+
+      addPoint(&kg, lon, lat + 1 / 3.0);
+      addPoint(&kg, lon, lat + 1 / 2.0);
+      addPoint(&kg, lon, lat + 2 / 3.0);
+
+      addPoint(&kg, lon + 1 / 3.0, lat);
+      addPoint(&kg, lon + 1 / 3.0, lat + 1 / 3.0);
+      addPoint(&kg, lon + 1 / 3.0, lat + 1 / 2.0);
+      addPoint(&kg, lon + 1 / 3.0, lat + 2 / 3.0);
+
+      addPoint(&kg, lon + 1 / 2.0, lat);
+      addPoint(&kg, lon + 1 / 2.0, lat + 1 / 3.0);
+      addPoint(&kg, lon + 1 / 2.0, lat + 1 / 2.0);
+      addPoint(&kg, lon + 1 / 2.0, lat + 2 / 3.0);
+
+      addPoint(&kg, lon + 2 / 3.0, lat);
+      addPoint(&kg, lon + 2 / 3.0, lat + 1 / 3.0);
+      addPoint(&kg, lon + 2 / 3.0, lat + 1 / 2.0);
+      addPoint(&kg, lon + 2 / 3.0, lat + 2 / 3.0);
+    }
+  }
+  return kg;
+}
+
+namespace computeResultTest {
+
+std::shared_ptr<QueryExecutionTree> buildIndexScan(QueryExecutionContext* qec,
+                                                   TripleComponent subject,
+                                                   std::string predicate,
+                                                   TripleComponent object) {
+  return ad_utility::makeExecutionTree<IndexScan>(
+      qec, Permutation::Enum::PSO, SparqlTriple{subject, predicate, object});
+}
+
+std::shared_ptr<QueryExecutionTree> buildJoin(
+    QueryExecutionContext* qec, std::shared_ptr<QueryExecutionTree> tree1,
+    std::shared_ptr<QueryExecutionTree> tree2, Variable joinVariable) {
+  auto varCol1 = tree1->getVariableColumns();
+  auto varCol2 = tree2->getVariableColumns();
+  size_t col1 = varCol1[joinVariable].columnIndex_;
+  size_t col2 = varCol2[joinVariable].columnIndex_;
+  return ad_utility::makeExecutionTree<Join>(qec, tree1, tree2, col1, col2,
+                                             true);
+}
+
+std::shared_ptr<QueryExecutionTree> buildMediumChild(
+    QueryExecutionContext* qec, std::string subject1_, std::string predicate1,
+    std::string object1_, std::string subject2_, std::string predicate2,
+    std::string object2_, std::string subject3_, std::string predicate3,
+    std::string object3_, std::string joinVariable1_,
+    std::string joinVariable2_) {
+  TripleComponent subject1{Variable{subject1_}};
+  TripleComponent subject2{Variable{subject2_}};
+  TripleComponent subject3{Variable{subject3_}};
+  TripleComponent object1{Variable{object1_}};
+  TripleComponent object2{Variable{object2_}};
+  TripleComponent object3{Variable{object3_}};
+  Variable joinVariable1{joinVariable1_};
+  Variable joinVariable2{joinVariable2_};
+  auto scan1 = buildIndexScan(qec, subject1, predicate1, object1);
+  auto scan2 = buildIndexScan(qec, subject2, predicate2, object2);
```

Comment:
Then you can write a lambda that takes a single of those triples and turns it into an index scan.

Reply:
Best define that function out of this function, you can reuse it in the `buildSmallChild` function.